### PR TITLE
Bfs dart

### LINKED ISSUE #2171: Breadth First Search
- [x] C
- [x] C++
- [x] C# - @prashansag62
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go - @Aju100
- [x] Java
- [x] JS
- [ ] Kotlin
- [ ] Php - @srimani-programmer
- [x] Python
- [x] Readme
- [x] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Rukmini-Meda**
Fixes #2171 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Breadth first search is implemented in Dart.
- Graph is implemented as adjacency matrix.
- Works well for both directed and undirected graphs.

#### Languages Used:
- Dart

#### Files Added:
- Breadth_First_Search.dart

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Breadth_First_Search/Breadth_First_Search.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        }
+    }
+    int source=0;
+
+    // Sample input graph
+    // 0 - 1 - 2
+    // |     /
+    // 3 - 4
+
+    graph[0][1]=1;
```
Comment (MastersAbh) [inline Breadth_First_Search/Breadth_First_Search.dart]: Please modify your code to take dynamic inputs.

File: Breadth_First_Search/Breadth_First_Search.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    int no_vertices=5;
+
+    // Graph is implemented as adjacency matrix.
+    var graph = new List.generate(no_vertices,(_)=>new List(no_vertices));
+    for(int i=0;i<no_vertices;i++){
+        for(int j=0;j<no_vertices;j++){
+            graph[i][j]=0;
+        }
+    }
+    int source=0;
```
Comment (MastersAbh) [inline Breadth_First_Search/Breadth_First_Search.dart]: source should also be dynamic

**Comment by Rukmini-Meda** [comment]
I have made all the requested changes and also added the functionality to choose between directed and undirected graphs. The code is ready to be merged. Please let me know if it still requires changes. Thank you.

File: Breadth_First_Search/Breadth_First_Search.dart [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      int source = int.parse(stdin.readLineSync());
+
+      print("Enter destination vertex:");
+      int destination = int.parse(stdin.readLineSync());
+
+      if(mode == "undirected"){
+        graph[source][destination] = 1;
+        graph[destination][source] = 1;
+      }
+
```
Comment (MastersAbh) [inline Breadth_First_Search/Breadth_First_Search.dart]: Remove this line between if else

File: Breadth_First_Search/Breadth_First_Search.dart [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  
+    print("Enter number of edges:");
+    var input1 = stdin.readLineSync();
+    int no_edges = int.parse(input);
+    print("Enter source and destination vertices of edges:\n");
+    print("Note: Enter source and destination vertices as 0,1,2,...");
+    
+    for(int i = 1; i <= no_edges; i++){
+      
+      print("Edge #$i:");
```
Comment (MastersAbh) [inline Breadth_First_Search/Breadth_First_Search.dart]: indentation inside for loop should be 4 spaces.

File: Breadth_First_Search/Breadth_First_Search.dart [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      
+      print("Enter source vertex:");
+      int source = int.parse(stdin.readLineSync());
+
+      print("Enter destination vertex:");
+      int destination = int.parse(stdin.readLineSync());
+
+      if(mode == "undirected"){
+        graph[source][destination] = 1;
```
Comment (MastersAbh) [inline Breadth_First_Search/Breadth_First_Search.dart]: indentation

**Comment by Rukmini-Meda** [comment]
I have made the changes. Please review and reply.

**Pull request merged**